spec/support/utilities/iis_application.rb

```diff
@@ -0,0 +1,20 @@
+def has_app(app_name)
+  !(on(default, "C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe -noninteractive -noprofile -executionpolicy bypass -command \"& {Get-WebApplication -Name #{app_name}}\"").stdout =~ /Started/i).nil?
+end
+
+def create_app(site_name, app_name, directory)
+  on(default, "C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe -noninteractive -noprofile -executionpolicy bypass -command \"& {New-WebApplication -Site #{site_name} -Name #{app_name} -PhysicalPath #{directory} -Force -ErrorAction Stop}\"") unless has_app(app_name)
+end
+
+def remove_app(app_name)
+  on(default, "C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe -noninteractive -noprofile -executionpolicy bypass -command \"& {Remove-WebApplication -Name #{app_name}}\"") if has_app(app_name)
+end
+
+def create_virtual_directory(site,name,directory)
+  on(default, "C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe -noninteractive -noprofile -executionpolicy bypass -command \"& {New-WebVirtualDirectory -Site #{site} -Name #{name} -physicalPath #{directory} -Force -ErrorAction Stop}\"")
+end
+
+def create_selfsigned_cert(dnsname)
+  result = on(default, "C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe -noninteractive -noprofile -executionpolicy bypass -command \"& {(New-SelfSignedCertificate -DnsName '#{dnsname}' -CertStoreLocation cert:Localmachine\\My).Thumbprint}\"")
```

Comment:
This will break on windows 2008. `New-SelfSignedCertificate` only came into existence on 2012. This is one of the issues I called out when we discussed adding ssl to the acceptance tests. We now have to add functions to create self-signed on 2008